internal/queryexecutor/executor.go

```diff
@@ -0,0 +1,160 @@
+package queryexecutor
+
+import (
+	"encoding/json"
+	"reflect"
+	"strings"
+
+	"github.com/IBM-Blockchain/bcdb-server/internal/worldstate"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/constants"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/logger"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+	"github.com/pkg/errors"
+)
+
+type WorldStateQueryExecutor struct {
+	db     worldstate.DB
+	logger *logger.SugarLogger
+}
+
+func NewWorldStateQueryExecutor(db worldstate.DB, l *logger.SugarLogger) *WorldStateQueryExecutor {
+	return &WorldStateQueryExecutor{
+		db:     db,
+		logger: l,
+	}
+}
+
+type attributesConditions map[string]*attrConditions
+
+type attrConditions struct {
+	valueType  types.Type
+	conditions map[string]interface{}
+}
+
+func (e *WorldStateQueryExecutor) validateAndDisectConditions(dbName string, conditions map[string]interface{}) (attributesConditions, error) {
+	// when we reach here, we assume that the given dbName exist
+	marshledIndexDef, _, err := e.db.GetIndexDefinition(dbName)
+	if err != nil {
+		return nil, err
+	}
+
+	if marshledIndexDef == nil {
+		return nil, errors.New("no index has been defined on the database " + dbName)
+	}
+
+	indexDef := map[string]types.Type{}
```

Comment:
Agree. As it might affect a few other packages, we can do this in a separate PR -- created an issue https://github.com/IBM-Blockchain/bcdb-server/issues/168